From 00d2b62575d1216ec8d949a791a6dfa31bdd730d Mon Sep 17 00:00:00 2001
From: Stephen Sherratt <stephen@sherra.tt>
Date: Tue, 27 Aug 2024 21:20:15 +1000
Subject: [PATCH] chore(nix): add experimental binaries with features enabled

Use `overrideAttrs` to change the behaviour and export
new target `dune-experimental` and `dune-static-experimental`

Co-authored-by: Etienne Millon <me@emillon.org>

Signed-off-by: Stephen Sherratt <stephen@sherra.tt>
Signed-off-by: Etienne Marais <dev@maiste.fr>
Signed-off-by: Etienne Millon <me@emillon.org>
---
 flake.nix | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/flake.nix b/flake.nix
index 3e805a9d0..ff04762d0 100644
--- a/flake.nix
+++ b/flake.nix
@@ -60,6 +60,14 @@
         dune-static-overlay
       ];
 
+      add-experimental-configure-flags = pkg: pkg.overrideAttrs {
+        configureFlags =
+          [
+            "--enable-toolchains"
+            "--enable-pkg-build-progress"
+          ];
+      };
+
       ocamlformat =
         let
           ocamlformat_version =
@@ -102,6 +110,8 @@
         };
         dune = self.packages.${system}.default;
         dune-static = pkgs-static.pkgsCross.musl64.ocamlPackages.dune;
+        dune-experimental = add-experimental-configure-flags self.packages.${system}.dune;
+        dune-static-experimental = add-experimental-configure-flags self.packages.${system}.dune-static;
       };
 
       devShells =
-- 
2.46.0

